### ISSUE TITLE
Fix S1168: Rule should not report when returning null within a lambda

### ISSUE THREAD

**Issue by Evangelink** [SonarSource/sonar-csharp#761]
### Description

Currently [the rule](https://jira.sonarsource.com/browse/RSPEC-1168) raises an issue when returning null from within a lambda declared inside of a method returning a collection.

### Repro steps

```
public List<int> Method()
{
    Func<int?> aFunc = () => {
        return null;
    };
    return new List<int>();
}
```

### Related information

* SonarLint for VS 3.5.0.1803
